hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -179,6 +179,12 @@ protected long getTotalStorageMemoryCost(K key, R record) {
     }
 
     protected boolean isRecordExpired(R record) {
+        if (!canUpdateStats(record)) {
+            // A record can only be checked for expiry if its record state is
+            // READ_PERMITTED. We can check reserved records for expiry.
```

Comment:
should read "We can't check reserved records"

Reply:
fixed.